athena-elasticsearch/src/main/java/com/amazonaws/athena/connectors/elasticsearch/ElasticsearchMetadataHandler.java

```diff
@@ -305,6 +315,31 @@ public GetSplitsResponse doGetSplits(BlockAllocator allocator, GetSplitsRequest
         return new GetSplitsResponse(request.getCatalogName(), splits);
     }
 
+    @Override
+    public GetDataSourceCapabilitiesResponse doGetDataSourceCapabilities(BlockAllocator allocator, GetDataSourceCapabilitiesRequest request)
```

Comment:
I'd prefer to not wrap the this call into another request; instead the best option here would be to extract the common code found in `doGetTable` into an internal private method and keep both APIs call separated.